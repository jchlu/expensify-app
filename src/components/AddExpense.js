import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { addExpense } from '../actions/expenses'

/* Avoid inline functions by switching to a class
* exporting the unconnected version to allow testing */
export class AddExpense extends React.Component {
  onSubmit = (expense) => {
    this.props.onSubmit(expense)
    this.props.history.push('/')
  }
  render() {
    return (
      <div>
      <p>Add Expense</p>
      <ExpenseForm
        onSubmit={this.onSubmit}
      />
    </div>
  )
  }
}

/* const AddExpense = (props) => (
  <div>
    <p>Add Expense</p>
    <ExpenseForm
      onSubmit={(expense) => {
        props.onSubmit(expense)
        props.history.push('/')
      }}
    />
  </div>
) */

const mapDispatchToProps = (dispatch) => ({
  // return dispatcher functions, abstracting them away from the component
  onSubmit: (expense) => dispatch(addExpense(expense))
})
export default connect(undefined, mapDispatchToProps)(AddExpense)
