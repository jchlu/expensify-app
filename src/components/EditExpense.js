import React from 'react'
// Internal Components

export default class EditExpense extends React.Component {
  state = {
    options: []
  }

  handleState = () => {
    this.setState(() => ({ options: [] }))
  }

  render() {
    const title = 'Edit Expense'
    return (
      <div>
        <div className="container">
          <p>{title}</p>
        </div>
      </div>
    )
  }
} // End of EditExpense class definition
