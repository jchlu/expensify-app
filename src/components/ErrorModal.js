import React from 'react'
// Third Party Components
import Modal from 'react-modal'

/** ShowErrorModal requires three params,
 * a message, a show boolean and a function to call on close */
const ErrorModal = (props) => (
  <Modal
    isOpen={!!props.showError}
    onRequestClose={props.handleClearShowError}
    contentLabel="Error Message"
    closeTimeoutMS={300}
    className="modal"
    appElement={document.getElementById('app')}
  >
    <h3 className="modal__title">Warning</h3>
    {props.errorMessage && <p className="modal__body">{props.errorMessage}</p>}
    <button className="button" onClick={props.handleClearShowError}>Okay</button>
  </Modal>
)

export default ErrorModal
