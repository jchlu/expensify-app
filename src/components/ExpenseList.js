import React from 'react'
import { connect } from 'react-redux'

const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    <p>{props.filters.text}</p>
    {props.expenses.length}
  </div>
)

/** What we want from the store to be passed as props */
const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
    filters: state.filters
  }
}

export default connect(mapStateToProps)(ExpenseList)

/*
// *** Original iteration: ***
const ConnectedExpenseList = connect((state) => {
  return {
    expenses: state.expenses
  }
})(ExpenseList)

export default ConnectedExpenseList

// *** Original becomes: ***
export default connect((state) => {
  return {
    expenses: state.expenses
  }
})(ExpenseList)

// *** Which then becomes: ***
const mapStateToProps = (state) => {
  return {
    expenses: state.expenses
  }
}
export default connect(mapStateToProps)(ExpenseList)
 */
