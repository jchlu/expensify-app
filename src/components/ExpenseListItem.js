import React from 'react'

export default ({ description, amount, createdAt }) => (
  <div>
    <p>{description}</p>
    <p>{amount} - {createdAt}</p>
  </div>
)
