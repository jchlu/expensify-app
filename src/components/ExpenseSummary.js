import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import getVisibleExpenses from '../selectors/expenses'
import summedExpenses from '../selectors/expenses-total'
import numeral from 'numeral'

const ExpenseSummary = (props) => {
  const count = props.expenses.length
  return (
    <div>
      {
        count > 0 &&
        <p>
          Viewing {count} Expense{count > 1 && 's'} Totalling {
            numeral(props.expensesTotal / 100).format('$0,0.00')
          }
        </p>
      }
    </div>
  )
}

ExpenseSummary.propTypes = {
  expenses: PropTypes.array
}

/** What we want from the store to be passed as props */
const mapStateToProps = (state) => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters),
    expensesTotal: summedExpenses(
      getVisibleExpenses(state.expenses, state.filters)
    )
  }
}

export default connect(mapStateToProps)(ExpenseSummary)
