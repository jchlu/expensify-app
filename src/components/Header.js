import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = (props) => {
  return (
    <div className="header">
      <h1 className="header__title">{props.title}</h1>
      <p>
        |&nbsp;<NavLink className="header__link" activeClassName="header__link--active" to="/" exact={true}>Go Home</NavLink>&nbsp;
          |&nbsp;<NavLink className="header__link" activeClassName="header__link--active" to="/create">Add Expense</NavLink>&nbsp;
          |&nbsp;<NavLink className="header__link" activeClassName="header__link--active" to="/help">Help</NavLink>&nbsp;|
      </p>
    </div>
  )
}

Header.defaultProps = {
  title: 'Expensify'
}
export default Header
