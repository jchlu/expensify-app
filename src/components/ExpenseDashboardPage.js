import React from 'react'
import ExpenseList from './ExpenseList'
// Internal Components

const ExpenseDashboardPage = () => (
  <div>
    <h1><ExpenseList /></h1>
  </div>
)

export default ExpenseDashboardPage
