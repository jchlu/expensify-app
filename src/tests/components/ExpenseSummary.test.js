import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('Should render summary with two fixtures', () => {
  const wrapper = shallow(<ExpenseSummary count={2} total={12345} />)
  expect(wrapper).toMatchSnapshot()
})

test('Should render summary with a single fixture', () => {
  const wrapper = shallow(<ExpenseSummary count={1} total={1234} />)
  expect(wrapper).toMatchSnapshot()
})
