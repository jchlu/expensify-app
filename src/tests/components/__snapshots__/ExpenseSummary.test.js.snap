// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render summary with a single fixture 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      Expense
       Totalling 
      <span>
        $12.34
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render summary with two fixtures 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      Expenses
       Totalling 
      <span>
        $123.45
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
